### DISCUSSION TITLE
Domain Masking?

### DISCUSSION THREAD

**Discussion by brunocrosier** [vercel/community#50] [Help] [answered]
Is it possible to have domain masking with Vercel ?

For example, if I'm hosting a website (example.com) where each user has an "profile page" (example.com/bruno)

Would the "bruno" user be able to forward bruno.com to example.com/bruno easily ?

**Comment by styfle** [comment]
Bruno could create a deployment with [`rewrites`](https://vercel.com/docs/configuration#project/rewrites)  in `vercel.json` and add the domain `bruno.com` to their project.

### vercel.json

```json
{
  "rewrites": [
    { "source": "(.*)", "destination": "https://example.com/bruno$1" }
  ]
}
```

**Comment by esteban89** [comment]
> Is it possible to have domain masking with Vercel ?
> 
> For example, if I'm hosting a website (example.com) where each user has an "profile page" (example.com/bruno)
> 
> Would the "bruno" user be able to forward bruno.com to example.com/bruno easily ?

Did you find any other way to do it using the same project? I'm trying different ways by adding a domain to the same project and trying to rewrite without any luck.

**Reply by steven-tey** [reply]
@esteban89 I just posted an example for this in my comment below, hope it helps!

**Comment by brunocrosier** [comment]
Hey @styfle thanks for the answer. I think I should have asked my question in a better way. I'm essentially asking the same as @esteban89 - in other words, can we point multiple domains (not managed in Vercel) to specific URLs in our main site? 

A good example would be something like Dan.com where you can put domain names for sale and point them to a specific nameserver so that they redirect to the correct Dan.com page when visited.

Eg. I put Bruno.com for sale on dan.com and therefore after I make some DNS changes to Bruno.com, it redirects to dan.com/landing-page/bruno

Is something like that possible?

**Reply by ralphilius** [reply]
Yes. You can add bruno.com to your project in the dashboard then point CNAME to Vercel's NS. This has 50 domains limit per project though and it would cost a few thousands dollars per month if you want unlimited domains.

**Reply by brunocrosier** [reply]
Thanks @ralphilius . The key point here though is that the domain is externally owned, so it couldn't be added into the Vercel dashboard in the way you describe. What I'm looking for is a way for the external domain to redirect to a specific page on the main site, without needing to go through the Vercel dashboard (ie. happens automatically)

**Reply by ralphilius** [reply]
What you want is not possible since Vercel needs to identify if the external domain is authorized to load your specific page. Otherwise, any domain that points to your project domain would serve the same content as others.

**Reply by brunocrosier** [reply]
> What you want is not possible since Vercel needs to identify if the external domain is authorized to load your specific page. Otherwise, any domain that points to your project domain would serve the same content as others.

thanks for explaining why it's not possible @ralphilius !

**Reply by royalcala** [reply]
Hi @brunocrosier I had ask to vercel for the price of increment the domains but I didn't receive an answer, do you find some alternative to do this and have more than 50 domains

**Comment by esteban89** [comment]
@brunocrosier I found a way to do it (but it's not pretty), Combining @styfle answer with Vercel API (https://vercel.com/docs/api) you can create a project, add a custom domain to it and upload the vercel.json file automatically from your code.

**Reply by brunocrosier** [reply]
🤯 that's genius!

**Comment by steven-tey** [comment, marked as answer]
Update (1/20/2022): we rolled out the full example here: https://demo.vercel.pub/platforms-starter-kit

Hey @brunocrosier! We recently rolled out a [simple example](https://github.com/vercel/examples/tree/main/edge-functions/hostname-rewrites) that allows you to do exactly that – create unique content pages with a multi-tenant infrastructure using [edge function](https://vercel.com/features/edge-functions) rewrites and [ISR](https://vercel.com/docs/concepts/next.js/incremental-static-regeneration). Each of your users gets assigned a unique username when they create their account, with the (usually paid) option to add a custom domain. We also helped [read.cv implement this](https://read.cv/support/Q00tMVbWBlA32mb9IKNn) for their site and it went really well!

**Reply by jensmeindertsma** [reply]
Is there any way to use these edge function rewrites without using Next.js?

**Reply by leerob** [reply]
Yes! We are working to add more examples here, but you currently start using them by placing a `_middleware.js` or `.ts` file in the root of your project directory.
